File: Linguagem de Programação I/Extras/Beecrowd_Extra/1023.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

let T = 0;
let input = [];

readline.on('line', (line) => {
    input.push(line);
});

readline.on('close', () => {
    for (let i = 0; i < input.length; i++) {
        let N = parseInt(input[i]);
        if (N === 0) break;

        if (T > 0) console.log();

        let totalX = 0, totalY = 0;
        let consumos = [];
        for (let j = 0; j < N; j++) {
            let [X, Y] = input[++i].split(' ').map(Number);
            totalX += X;
            totalY += Y;
            let key = Math.floor(Y / X);
            let found = false;
            for (let pair of consumos) {
                if (pair.key === key) {
                    pair.value += X;
                    found = true;
                    break;
                }
            }
            if (!found) {
                consumos.push({ key, value: X });
            }
        }

        T++;
        console.log(`Cidade# ${T}:`);

        consumos.sort((a, b) => a.key - b.key);
        for (let pair of consumos) {
            console.log(`${pair.value}-${pair.key} `);
        }

        let consumo_medio = Math.floor((100.0 * totalY) / totalX) / 100;
        console.log(`Consumo medio: ${consumo_medio.toFixed(2)} m3.`);
    }
});

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1023.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

struct Pair {
    int key;
    int value;
};

bool compare(const Pair &a, const Pair &b) {
    return a.key < b.key;
}

int main() {
    int T = 0;
    int N;
    while (cin >> N && N != 0) {
        if (T > 0) {
            cout << "\n";
        }

        int totalX = 0, totalY = 0;
        vector<Pair> consumos;
        for (int i = 0; i < N; i++) {
            int X, Y;
            cin >> X >> Y;
            totalX += X;
            totalY += Y;
            int key = Y / X;
            auto it = find_if(consumos.begin(), consumos.end(), [key](const Pair &p) { return p.key == key; });
            if (it != consumos.end()) {
                it->value += X;
            } else {
                consumos.push_back({key, X});
            }
        }

        T += 1;
        cout << "Cidade# " << T << ":\n";

        sort(consumos.begin(), consumos.end(), compare);
        for (const auto &p : consumos) {
            cout << p.value << "-" << p.key << " ";
        }
        cout << "\n";

        double consumo_medio = floor((100.0 * totalY) / totalX) / 100;
        printf("Consumo medio: %.2f m3.\n", consumo_medio);
    }
    return 0;
}